transform/instance.go

```diff
@@ -107,13 +111,58 @@ func (at *arrayTransformer) baseValue(in interface{}, path string, modifier path
 	return nil, false, nil
 }
 
+func (at *arrayTransformer) baseValueXML(in interface{}, path string, modifier pathModifier) ([]interface{}, bool, error) {
+	// 1. Use a transform if it exists
+	if at.transforms != nil {
+		rawValue, err := at.transforms.transform(in, "array", modifier, at.format)
+		if err != nil {
+			return nil, false, err
+		}
+
+		//if rawValue is an array of xml nodes we need to append them to newValue for return as []interface{}
+		xmlNodeArray, ok := rawValue.([]*xmlquery.Node)
+		if ok {
+			newValue := make([]interface{}, len(xmlNodeArray))
+			for i, item := range xmlNodeArray {
+				newValue[i] = item
+			}
+			return newValue, false, nil
+		}
+
+		if rawValue != nil {
+			newValue, ok := rawValue.([]interface{})
+			if !ok {
+				newValue = []interface{}{rawValue}
+			}
+			return newValue, true, nil
+		}
+	}
+
+	// 2. Fall back to the JSON Schema default value.
+	if at.defaultValue != nil {
+		return at.defaultValue, true, nil
+	}
+	return nil, false, nil
+}
+
+// baseValue routes to the correct arrayTransformer.baseValue format
+func (at *arrayTransformer) baseValue(in interface{}, path string, modifier pathModifier) ([]interface{}, bool, error) {
+	if at.format == jsonInput {
```

Comment:
I seem to ask this question a lot, but is it worth it to create a map for these? a `map[inputFormat]baseValueFunc`? Hypothetically we only will be doing XML and JSON is just a thought